redis/Dockerfile

```diff
@@ -13,34 +13,141 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# https://hub.docker.com/_/redis
-FROM redis:7.4.2
-LABEL maintainer="Astronomer <humans@astronomer.io>"
+# https://hub.docker.com/r/curlimages/curl/tags
+FROM curlimages/curl:latest AS downloader
+
+ENV REDIS_VERSION=7.4.2
+ENV REDIS_DOWNLOAD_URL=http://download.redis.io/releases/redis-7.4.2.tar.gz
+ENV REDIS_DOWNLOAD_SHA=4ddebbf09061cbb589011786febdb34f29767dd7f89dbe712d2b68e808af6a1f
+
+WORKDIR /tmp
 
-# Update packages to solve security vulnerabilities
-ENV DEBIAN_FRONTEND=noninteractive
-RUN apt-get update && \
-    apt-get dist-upgrade -y && \
-    rm -rf /var/lib/apt/lists/*
+RUN mkdir redis ; curl -fsSL $REDIS_DOWNLOAD_URL -o redis.tar.gz; echo "$REDIS_DOWNLOAD_SHA *redis.tar.gz" | sha256sum -c -;	\
+    tar -xzf redis.tar.gz -C redis --strip-components=1; \
+    rm redis.tar.gz;
+
+# https://raw.githubusercontent.com/redis/docker-library-redis/refs/heads/master/7.4/debian/Dockerfile
+FROM debian:bookworm-slim
```

Comment:
All the build steps in this Dockerfile should be put into a shell script. The build steps should go into a build stage that is not the final stage of the Dockerfile.